### ISSUE TITLE
Disabled keys not passing isDisable prop to item

### ISSUE THREAD

**Issue by lucasmotta** [adobe/react-spectrum#1316]
# 🐛 Bug Report 

I've been following the example on the [useMenu](https://react-spectrum.adobe.com/react-aria/useMenu.html) documentation, but I cannot get the `disabledKeys` prop to work properly. The keyboard navigation seems to respect the disabledKeys, but the Item itself receives `undefined` when accessing `item.isDisabled`.

## 🤔 Expected Behavior

The menu item should have the correct `isDisabled` state - either true or false, but not undefined.

## 😯 Current Behavior

When accessing `item.isDisabled`, as written at the documentation, it returns undefined.

## 💻 Code Sample

Here's a Codesandbox to demonstrate the issue:

https://codesandbox.io/s/sweet-bash-ywhx4?file=/src/Menu.js:1041-1067

**Comment by LFDanLu** [comment]
@lucasmotta I believe this is bug in the docs, based off a time when the `Node` objects in a collection contained things like `isDisabled`, `isSelected`, etc. The docs should probably follow the approach we use for our own [`MenuItem`](https://github.com/adobe/react-spectrum/blob/main/packages/@react-spectrum/menu/src/MenuItem.tsx#L54) where we look at the `state.disabledKeys` to see if a item should be disabled.

Thanks for catching this!

**Comment by razvanip** [comment]
@LFDanLu, seems to be a bug in the docs.
A fix should include:
- docs update
- or adding support for `isDisabled` in `useMenuItem` ?

**Comment by razvanip** [comment]
sure, i can pick it up.

**Comment by LFDanLu** [comment]
@razvanip Sounds good!
